url7a69e77641_benmouh_MySgbd_tgz-pJ8-Demo_CheckRowNumJ8/src/org/sgbd/Mysgbd/parser/MysgbdParser.java

```diff
@@ -0,0 +1,3669 @@
+    e1 = SQLExpotentExpression();
+    label_14:
+    while (true) {
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case 103:
+      case 107:
+        ;
+        break;
+      default:
+        jj_la1[88] = jj_gen;
+        break label_14;
+      }
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case 103:
+        op = jj_consume_token(103);
+        break;
+      case 107:
+        op = jj_consume_token(107);
+        break;
+      default:
+        jj_la1[89] = jj_gen;
+        jj_consume_token(-1);
+        throw new ParseException();
+      }
+      e2 = SQLExpotentExpression();
+      e = new MExpression(op.toString(), e1);
+      e.addOperand(e2);
+      e1 = e;
+    }
+      {if (true) return e1;}
+    throw new Error("Missing return statement in function");
+  }
+
+// SQLExpotentExpression ::= SQLUnaryExpression (** SQLUnaryExpression)*
+  static final public MExp SQLExpotentExpression() throws ParseException {
+  Token op;
+  MExp e1, e2;
+  MExpression e = null;
+  boolean single = true;
+    e1 = SQLUnaryExpression();
+    label_15:
+    while (true) {
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case 108:
+        ;
+        break;
+      default:
+        jj_la1[90] = jj_gen;
+        break label_15;
+      }
+      op = jj_consume_token(108);
+      e2 = SQLUnaryExpression();
+      if(single) e = new MExpression(op.toString(), e1);
+      single = false;
+      e.addOperand(e2);
+    }
+      {if (true) return (single ? e1 : e);}
+    throw new Error("Missing return statement in function");
+  }
+
+// SQLUnaryExpression ::= [+|-] SQLPrimaryExpression
+  static final public MExp SQLUnaryExpression() throws ParseException {
+  Token op = null;
+  MExp e1, e;
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case 101:
+    case 102:
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case 101:
+        op = jj_consume_token(101);
+        break;
+      case 102:
+        op = jj_consume_token(102);
+        break;
+      default:
+        jj_la1[91] = jj_gen;
+        jj_consume_token(-1);
+        throw new ParseException();
+      }
+      break;
+    default:
+      jj_la1[92] = jj_gen;
+      ;
+    }
+    e1 = SQLPrimaryExpression();
+    if(op == null) e = e1;
+    else e = new MExpression(op.toString(), e1);
+    {if (true) return e;}
+    throw new Error("Missing return statement in function");
+  }
+
+// 01/07/00 PY.Gibello - Added COUNT(*) to SQLPrimaryExpression
+// TBD should be MConstant("COUNT(*)", MConstant.STRING))
+// instead of MConstant("*", MConstant.COLUMNNAME) ??
+// 01/06/02 PY.Gibello - Added ALL | DISTINCT (then the column name
+// starts with "all " or "distinct " when required, which may not be the
+// fairest solution...)
+
+// SQLPrimaryExpression ::= NULL | OuterJoinExpression
+//   | AggregateFunc ( [ ALL | DISTINCT ] TableColumn ) | TableColumn
+//   | COUNT(*)
+//   | <NUMBER> | <STRING> | <BIND>
+  static final public MExp SQLPrimaryExpression() throws ParseException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 23. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.